### ISSUE TITLE
FIR conversion to LLVM for fir.box_dim produces invalid IR

### ISSUE THREAD

**Issue by ftynse** [llvm/llvm-project#53064]
The conversion of

```
func @extract_dims(%arg0: !fir.box<!fir.array<*:f64>>) -> index {
  %c1 = arith.constant 0 : i32
  %0:3 = fir.box_dims %arg0, %c1 : (!fir.box<!fir.array<*:f64>>, i32) -> (index, index, index)
  return %0 : index
}
```

to the LLVM dialect produces IR that accesses a struct with invalid index (7 for a struct of 7 elements),

This code was present in the `convert-to-llvm.fir` test. https://github.com/llvm/llvm-project/commit/301226562b28088b112651cadf6695eba1e2ef87 disabled it as it triggered assertion in MLIR.

**Comment by schweitzpgi** [comment]
First take is that this does not appear to be a valid test.

**Comment by ftynse** [comment]
Looks fixed by 7ecce1f6959f8f196cba7f0b9411fa5d50e17b99
